bot/commands/my_stats.py

```diff
@@ -0,0 +1,256 @@
+
+    async def send_graphs_via_dm(
+        self,
+        user: discord.User,
+        graph_files: List[str]
+    ) -> None:
+        """Send graphs to user via DM with proper error handling.
+        
+        Parameters
+        ----------
+        user : discord.User
+            The user to send graphs to
+        graph_files : List[str]
+            List of graph file paths
+            
+        Raises
+        ------
+        DMError
+            If sending DMs fails
+        """
+        try:
+            dm_channel = await user.create_dm()
+            
+            for graph_file in graph_files:
+                try:
+                    # Extract just the filename without the path
+                    filename = os.path.basename(graph_file)
+                    
+                    if not os.path.exists(graph_file):
+                        logging.error(f"Graph file not found: {filename}")
+                        continue
+
+                    await dm_channel.send(file=discord.File(graph_file))
+                    logging.info(
+                        self.translations["log_sending_graph_file"].format(
+                            file=filename
+                        )
+                    )
+                except discord.HTTPException as e:
+                    logging.error(
+                        f"Failed to send graph {filename}: {str(e)}"
+                    )
+                    raise DMError(f"Failed to send graph: {filename}") from e
+                    
+        except discord.Forbidden as e:
+            logging.warning(f"Cannot send DM to user: {str(e)}")
+            raise DMError("DMs are disabled") from e
+        except Exception as e:
+            logging.error(f"Unexpected error while sending DMs: {str(e)}")
+            raise DMError("Failed to send graphs via DM") from e
+
+    @app_commands.command(
+        name="my_stats",
+        description="Get your personal Plex statistics"
+    )
+    @app_commands.describe(email="Your Plex email address")
+    async def my_stats(self, interaction: discord.Interaction, email: str) -> None:
+        """Get your personal Plex statistics.
+
+        Parameters
+        ----------
+        interaction : discord.Interaction
+            The interaction instance
+        email : str
+            The user's Plex email address
+        """
+        if not await self.check_cooldowns(
+            interaction,
+            self.config["MY_STATS_COOLDOWN_MINUTES"],
+            self.config["MY_STATS_GLOBAL_COOLDOWN_SECONDS"]
+        ):
+            return
+
+        await interaction.response.defer(ephemeral=True)
+
+        try:
+            # Validate email input
+            if not email or '@' not in email:
+                await interaction.followup.send(
+                    "Please provide a valid email address.",
+                    ephemeral=True
+                )
+                return
```

Comment:
It'll be complicated to do this. I'm guessing one approach could be using a SMTP server-like setup, instead of relying on Discord. But since it's just statistics, with no real information on what the user have watched, my verdict is that it's pretty harmless.

But it's a thing to consider, but for now I'm not going to do anything about it.